allennlp/models/dependency_parser.py

```diff
@@ -0,0 +1,464 @@
+from typing import Dict, Optional, Tuple
+
+from overrides import overrides
+import torch
+import torch.nn.functional as F
+import numpy
+
+from allennlp.common.checks import check_dimensions_match, ConfigurationError
+from allennlp.data import Vocabulary
+from allennlp.modules import Seq2SeqEncoder, TextFieldEmbedder, Embedding
+from allennlp.modules.biaffine_attention import BiaffineAttention
+from allennlp.models.model import Model
+from allennlp.nn import InitializerApplicator, RegularizerApplicator
+from allennlp.nn.util import get_text_field_mask, get_range_vector
+from allennlp.nn.util import get_device_of, last_dim_log_softmax, get_lengths_from_binary_sequence_mask
+from allennlp.nn.decoding.chu_liu_edmonds import decode_mst
+from allennlp.training.metrics import AttachmentScores
+
+
+@Model.register("dependency_parser")
+class DependencyParser(Model):
```

Comment:
do we really want to call this `DependencyParser`? that feels like the task name rather than the model name. (I am not sure, I just know that we're trying to be crisper about distinguishing between tasks and models)

also, do you want to add a comment (not sure if class-level or module-level) about which particular model this is, with a link to the paper and so on